### ISSUE TITLE
Option to disable automatic conversion

### ISSUE THREAD

**Issue by simonbyrne** [ajkeller34/Unitful.jl#63]
There are some cases where I would like to disable automatic conversion of units, and instead throw an error. Is this something that you would consider? I don't actually know how one would go about implementing it though.

A related issue is when there isn't a fixed conversion between units, e.g. currencies. Is there a way I could define new units that would simply throw an error if I attempted to use them in an incompatible way?

**Comment by ajkeller34** [comment]
Out of curiosity, do you have an example for when you would prefer to turn off automatic conversion? Right now, the conversion happens through promotion when the quantity types aren't exactly the same, which makes it tricky to work around. That's not to say there isn't a way to do it, but it would take some effort. I suppose you could try pirating the addition fallbacks to throw errors instead of promoting.

Regarding currencies, I haven't thought about it too carefully before. At first glance you might be able to get away with defining a new dimension for each currency, and then you could define units for that currency (Say you define USD as a dimension, then as units: dollars, quarters, dimes, nickels, cents.). Attempts to convert between currencies would throw a `DimensionError`, which is not the most helpful error in this case perhaps but would still prevent the conversion.

**Comment by simonbyrne** [comment]
>  At first glance you might be able to get away with defining a new dimension for each currency, and then you could define units for that currency.

That seems to work reasonably, though it means I can no longer dispatch on a "currency" dimensioned quantity.

**Comment by ajkeller34** [comment]
Unfortunately, I don't think it's currently possible to do dispatch on currency dimensioned quantities in usual way. However, if you aren't defining new currencies dynamically / you have a list of all possible currencies, you could do the following (using length and area as an example):

```
julia> using Unitful, Compat

julia> @compat LengthOrArea{T,U} = Quantity{T,<:Union{typeof(Unitful.𝐋),typeof(Unitful.𝐋^2)},U}
Unitful.Quantity{T,#s1,U} where #s1<:Union{Unitful.Dimensions{(Unitful.Dimension{:Length}(1//1),)}, Unitful.Dimensions{(Unitful.Dimension{:Length}(2//1),)}} where U where T

julia> isa(1u"m", LengthOrArea)
true

julia> isa(1u"m^2", LengthOrArea)
true

julia> isa(1u"m^3", LengthOrArea)
false
```

In your case you'd define dimensions for USD, GBP, etc. and take the union of the types of all those. Maybe this isn't so bad since it's not every day a new currency appears, but I confess it is not ideal.

Regarding automatic conversion, can you provide more details on when you need it to be turned off? Would you ideally want an error to be thrown every time `promote` is called because of a unit mismatch (in addition, subtraction, comparisons, etc.)?

**Comment by ajkeller34** [comment]
On a local branch I can disable automatic conversion and it works pretty well. It is part of a larger set of mostly non-breaking changes that I'll tag as 0.2.0 soon, probably after I finish up a few bug fixes and tag 0.1.5.

**Comment by ajkeller34** [comment]
Check out `FixedUnits` in the docs [here](http://ajkeller34.github.io/Unitful.jl/stable/conversion/) for a way to disable automatic conversion.
